### DISCUSSION TITLE
PorousFlow Density driven flow

### DISCUSSION THREAD

**Discussion by burritotukituki** [idaholab/moose#26887] [Q&A Modules: Porous Flow] [answered]
Hi,

I am interested in studying density-driven simulations. I thought a good starting point would be to utilize BrineFluidProperties with PorousFlow Brine. The concept involves mixing water with brine by maintaining a fixed pressure at a boundary.

I have a small model with water where I maintain a constant concentration of NaCl at the top. Please find my PF script attached.

https://github.com/burritotukituki/PF_concentration_density/tree/main

I've encountered difficulties in achieving convergence with this model. I'm unsure why the model isn't converging after a few time steps. Perhaps it's due to the density-driven flux resulting in small velocities that are challenging to simulate?

Any assistance or advice would be appreciated.

**Comment by 1runer** [comment]
Hi,

I guess you are aware of the [example of lava lamp](https://mooseframework.inl.gov/modules/porous_flow/lava_lamp.html)?
Or in terms of temperature-driven convection of [this example?](https://mooseframework.inl.gov/modules/porous_flow/natural_convection.html)

I would recommend to follow the example threating the mixture as 2phase - 2 component and solving individual equations for both phases. Convective mixing is highly sensitive to the mesh resolution. I think you should consider refining the mesh in the affected areas, either manually or using automatic mesh refinement (as shown in the examples above).

**Reply by burritotukituki** [reply]
Yeah I based my simulation on those examples. I will take a second look, maybe I am missing something! Thanks

**Reply by cpgr** [reply, marked as answer]
I haven't had a chance to check, but you might need to swap fluid_component indices in the kernels. When you pass Xnacl to the mass fraction material, you are saying that it is the mass fraction of fluid component 0 in phase 0, whereas you have it the other way around in your kernels.

Also, you should add diffusive kernels. Initially, mass transfer from the top boundary is via diffusion due to the gradient in mass fraction. As the instability grows, convective flux dominates and mass transfer is accelerated.

**Comment by burritotukituki** [comment]
Thanks for your replies!! It worked when I changed the indices. Thanks!!
